File: push/src/main/java/com/xiaomi/push/service/XmPushActionOperator.java
package com.xiaomi.push.service;

import com.nihility.XMPushUtils;
import com.xiaomi.xmpush.thrift.XmPushActionContainer;

public class XmPushActionOperator {
    private final XMPushService xmPushService;

    public XmPushActionOperator(XMPushService xmPushService) {
        this.xmPushService = xmPushService;
    }

    public void sendMessage(XmPushActionContainer sendMsgContainer, String packageName) {
        byte[] msgBytes = XMPushUtils.packToBytes(sendMsgContainer);
        xmPushService.sendMessage(packageName, msgBytes, false);
    }

}

File: push/src/main/java/com/xiaomi/push/service/XmPushActionOperator.kt
package com.xiaomi.push.service

import com.magisk317.push.hook.HookTraceCompat
import com.magisk317.service.XMPushServiceLifecycleBridge
import com.magisk317.XMPushUtils
import com.xiaomi.xmpush.thrift.XmPushActionContainer

class XmPushActionOperator(
    private val xmPushService: XMPushService
) {
    fun sendMessage(sendMsgContainer: XmPushActionContainer, packageName: String) {
        XMPushServiceLifecycleBridge.ensureCreated(xmPushService)
        val msgBytes = XMPushUtils.packToBytes(sendMsgContainer)
        HookTraceCompat.onSendMessage(packageName, msgBytes.size)
        xmPushService.sendMessage(packageName, msgBytes, false)
    }
}
